### DISCUSSION TITLE
Supporting older versions of Kafka when using latest version of Strimzi

### DISCUSSION THREAD

**Discussion by AvihuHenya** [strimzi/strimzi-kafka-operator#6816] [Q&A]
Hi,
I'm using Strimzi version 0.22.1 on Openshift 4.5 and I'm trying to upgrade the operator to version 0.28.0.
The problem is that I have Kafka clusters which are using 2.5.0 and 2.7.0 Kafka versions, and I cannot upgrade some of the Kafka clusters' versions (version property in Kafka CR) because of internal issues.

Because of that problem, I want to enable the option (keep some of the clusters in older versions) to support older versions of Kafka even though the operator doesn't support those old specific versions. 
I consider replacing the "version" property to "image" in all of the Kafka CRs and in each CR to set the image url according to the desired Kafka version of the clusters.

When I tried to make that changes I got errors in the Zookeeper Statefulset pod logs about "incorrect keystore password", and in the Kafka Statefulset pod logs: "Waiting for TLS sidecar to get ready/TLS sidecar is not ready yet", so the experiment didn't go well.

I will be glad if someone could help me understand if Strimzi supports that kind of upgrading, and if it does, what is the right way to do that.

Thanks in advance.

**Comment by scholzj** [comment]
Each Strimzi version supports only selected Kafka versions. So you will usually need to upgrade your Kafka cluster as well when upgrading Strimzi. For more details about our policy for supporting different Kafka versions, please see https://github.com/strimzi/strimzi-kafka-operator/blob/main/KAFKA_VERSION_SUPPORT.md
